test/test_fourier.jl

```diff
@@ -0,0 +1,407 @@
+    end
+
+    @testset "test_raw_high_bias" begin
+        C = [12986.0 + 8694.0im]
+        P1 = [476156.0]
+        P2 = [482751.0]
+        P1noise = 495955
+        P2noise = 494967
+    
+        coh = raw_coherence(C, P1, P2, P1noise, P2noise, 499; intrinsic_coherence=1)
+        coh_sngl = raw_coherence(C[1], P1[1], P2[1], P1noise, P2noise, 499; intrinsic_coherence=1)
+        @test coh==real(C .* conj(C))./ (P1 .* P2)
+        @test coh_sngl==real(C .* conj(C))[1] / (P1[1] * P2[1])
+    end
+    
+end
+
+@testset "test_fourier" begin
+    dt = 1
+    len = 100
+    ctrate = 10000
+    N = Int64(len / dt)
+    dt = len / N
+    times = sort(rand(Uniform(0, len), len * ctrate))
+    gti = [[0 len];;]
+    bins = LinRange(0, len, N + 1)
+    counts = fit(Histogram, times, bins).weights
+    errs = fill!(similar(counts),1) * sqrt(ctrate)
+    bin_times = (bins[1:end-1] + bins[2:end]) / 2
+    segment_size = 20.0
+    times2 = sort(rand(Uniform(0, len), len * ctrate))
+    counts2 = fit(Histogram, times2, bins).weights
+    errs2 = fill!(similar(counts2),1) * sqrt(ctrate)
+
+    @test get_average_ctrate(times, gti, segment_size) == ctrate
+    @test get_average_ctrate(bin_times, gti, segment_size; counts=counts) == ctrate
+
+    @testset "test_fts_from_segments_cts_and_events_are_equal" begin
+        N = Int64(round(segment_size / dt))
+        fts_evts = [
+            f for f in get_flux_iterable_from_segments(times, gti, segment_size, n_bin=N)
+        ]
+        fts_cts = [
+            f
+            for f in get_flux_iterable_from_segments(
+                bin_times, gti, segment_size, fluxes=counts
+            )
+        ]
+        for (fe, fc) in zip(fts_evts, fts_cts)
+            @test fe==fc
+        end
+    end
+
+    @testset "test_avg_pds_bad_input" begin
+        _times = rand(Uniform(0,1000),1)
+        out_ev = avg_pds_from_events(_times, gti, segment_size, dt,silent = true)
+        @test isnothing(out_ev)
+    end
+
+    @testset "test_avg_cs_bad_input" for return_auxil in [true, false]
+        _times1 = rand(Uniform(0,1000),1)
+        _times2 = rand(Uniform(0,1000),1)
+        out_ev = avg_cs_from_events(_times1, _times2, gti,
+                                    segment_size, dt, silent = true, return_auxil=return_auxil)
+        @test isnothing(out_ev) 
+    end
+
+    @testset "test_avg_pds_use_common_mean_similar_stats" for norm in ["frac", "abs", "none", "leahy"]
+        out_comm = avg_pds_from_events(
+            times,
+            gti,
+            segment_size,
+            dt,
+            norm=norm,
+            use_common_mean=true,
+            silent=true,
+            fluxes=nothing,
+        ).power
+        out = avg_pds_from_events(
+            times,
+            gti,
+            segment_size,
+            dt,
+            norm=norm,
+            use_common_mean=false,
+            silent=true,
+            fluxes=nothing,
+        ).power
+        @test std(out_comm)≈std(out) rtol=0.1
```

Comment:
Relative tolerance of 10% sounds fairly bad.  Is this the precision within which the two functions are supposed to agree?